x-pack/plugins/synthetics/public/plugin.ts

```diff
@@ -154,15 +156,15 @@ export class UptimePlugin
             defaultMessage: 'Overview',
           }),
           path: '/',
-          navLinkStatus: AppNavLinkStatus.visible,
+          navLinkStatus: plugins?.serverless ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden,
```

Comment:
I wonder if we should follow what APM implemented:
https://github.com/cauemarcondes/kibana/blob/1b462213ec5adffe40d78f0bdf59f1f98693d9cf/x-pack/plugins/apm/public/plugin.ts#L383

Reply:
APM also doesn't have a serverless dependency.

Reply:
We can, it will require a little additional revision to this file.

Reply:
Ah I see, it relies on the `xpack.apm.serverless.enabled` config.

Maybe following infra won't be that much of a change? https://github.com/cauemarcondes/kibana/blob/1b462213ec5adffe40d78f0bdf59f1f98693d9cf/x-pack/plugins/infra/public/plugin.ts#L67

FYI: I actually see many plugins following this pattern. But I'm no expert in serverless.

Reply:
I think effectively there's no difference, just an implementation detail. When Kibana's running in stateful mode, the setup/start plugins are undefined. When running in serverless, the setup plugin is set to `{}` and the start plugin contains some APIs to consume.

I'm already utilizing this in https://github.com/elastic/kibana/pull/176947, which is the reason I included the same logic here. When it comes to setting a flag for usage downstream in the Synthetics plugin, I am fine with picking it up from the config env as shown here, if that's the way most people are doing it.

Reply:
Implemented: [4fb1467](https://github.com/elastic/kibana/pull/177040/commits/4fb1467d0c4ec861fb060864667b4eced077d310).

Went back and tested this on serverless/stateful, seems to work fine.